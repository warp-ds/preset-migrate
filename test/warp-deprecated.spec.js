import { setup } from "./_helpers.js";
import { describe, expect, test, vi } from "vitest";

setup();


describe("deprecated warp classes", () => {
  test("Emits a warning if semantic color classes used with -default suffix are found", async (t) => {
    const warnSpy = vi.spyOn(global.console, 'warn')

    const classes = [
      's-color-background-default',
      's-color-background-subtle-default',
      's-color-background-primary-default',
      's-color-background-positive-default',
      's-color-background-negative-default',
      's-color-background-warning-default',
      's-color-background-info-default',
      's-color-border-default',
      's-color-border-primary-default',
      's-color-border-primary-subtle-default',
      's-color-border-positive-default',
      's-color-border-positive-subtle-default',
      's-color-border-negative-default',
      's-color-border-negative-subtle-default',
      's-color-border-warning-default',
      's-color-border-warning-subtle-default',
      's-color-border-info-default',
      's-color-border-info-subtle-default',
      's-color-icon-default',
      's-color-icon-subtle-default',
      's-color-text-default'
    ]

    const { css } = await t.uno.generate(classes);

    expect(css).toMatchInlineSnapshot('""');
    expect(warnSpy).toHaveBeenCalledTimes(classes.length);
    expect(warnSpy.calls.flat()).toMatchSnapshot();
  });

  test("Emits a warning if color classes without s- suffix are found", async (t) => {
    const warnSpy = vi.spyOn(global.console, 'warn')

    const classes = [
    'color-focused',
    'color-background',
    'color-background-subtle',
    'color-background-interactive',
    'color-background-interactive-hover',
    'color-background-interactive-selected',
    'color-text',
    'color-text-subtle',
    'color-text-placeholder',
    'color-text-inverted',
    'color-text-inverted-subtle',
    'color-text-link',
    'color-text-link-hover',
    'color-text-link-visited',]

    const { css } = await t.uno.generate(classes);

    expect(css).toMatchInlineSnapshot('""');
    expect(warnSpy).toHaveBeenCalledTimes(classes.length);
    expect(warnSpy.calls.flat()).toMatchSnapshot();
  });
})