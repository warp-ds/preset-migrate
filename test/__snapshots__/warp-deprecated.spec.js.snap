// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`deprecated warp classes > Emits a warning if color classes without s- suffix are found 2`] = `
[
  "[93m[DEPRECATED][0m color-focused -> use 's-focused'",
  "[93m[DEPRECATED][0m color-background -> see https://warp-ds.github.io/css-docs/background-color to find suitable semantic replacement",
  "[93m[DEPRECATED][0m color-background-subtle -> see https://warp-ds.github.io/css-docs/background-color to find suitable semantic replacement",
  "[93m[DEPRECATED][0m color-background-interactive -> see https://warp-ds.github.io/css-docs/background-color to find suitable semantic replacement",
  "[93m[DEPRECATED][0m color-background-interactive-hover -> see https://warp-ds.github.io/css-docs/background-color to find suitable semantic replacement",
  "[93m[DEPRECATED][0m color-background-interactive-selected -> see https://warp-ds.github.io/css-docs/background-color to find suitable semantic replacement",
  "[93m[DEPRECATED][0m color-text -> see https://warp-ds.github.io/css-docs/text-color to find suitable semantic replacement",
  "[93m[DEPRECATED][0m color-text-subtle -> see https://warp-ds.github.io/css-docs/text-color to find suitable semantic replacement",
  "[93m[DEPRECATED][0m color-text-placeholder -> see https://warp-ds.github.io/css-docs/text-color to find suitable semantic replacement",
  "[93m[DEPRECATED][0m color-text-inverted -> see https://warp-ds.github.io/css-docs/text-color to find suitable semantic replacement",
  "[93m[DEPRECATED][0m color-text-inverted-subtle -> see https://warp-ds.github.io/css-docs/text-color to find suitable semantic replacement",
  "[93m[DEPRECATED][0m color-text-link -> see https://warp-ds.github.io/css-docs/text-color to find suitable semantic replacement",
  "[93m[DEPRECATED][0m color-text-link-hover -> see https://warp-ds.github.io/css-docs/text-color to find suitable semantic replacement",
  "[93m[DEPRECATED][0m color-text-link-visited -> see https://warp-ds.github.io/css-docs/text-color to find suitable semantic replacement",
]
`;

exports[`deprecated warp classes > Emits a warning if semantic color classes used with -default suffix are found 2`] = `
[
  "[93m[DEPRECATED][0m s-color-background-default -> use 's-color-background'",
  "[93m[DEPRECATED][0m s-color-background-subtle-default -> use 's-color-background-subtle'",
  "[93m[DEPRECATED][0m s-color-background-primary-default -> use 's-color-background-primary'",
  "[93m[DEPRECATED][0m s-color-background-positive-default -> use 's-color-background-positive'",
  "[93m[DEPRECATED][0m s-color-background-negative-default -> use 's-color-background-negative'",
  "[93m[DEPRECATED][0m s-color-background-warning-default -> use 's-color-background-warning'",
  "[93m[DEPRECATED][0m s-color-background-info-default -> use 's-color-background-info'",
  "[93m[DEPRECATED][0m s-color-border-default -> use 's-color-border'",
  "[93m[DEPRECATED][0m s-color-border-primary-default -> use 's-color-border-primary'",
  "[93m[DEPRECATED][0m s-color-border-primary-subtle-default -> use 's-color-border-primary-subtle'",
  "[93m[DEPRECATED][0m s-color-border-positive-default -> use 's-color-border-positive'",
  "[93m[DEPRECATED][0m s-color-border-positive-subtle-default -> use 's-color-border-positive-subtle'",
  "[93m[DEPRECATED][0m s-color-border-negative-default -> use 's-color-border-negative'",
  "[93m[DEPRECATED][0m s-color-border-negative-subtle-default -> use 's-color-border-negative-subtle'",
  "[93m[DEPRECATED][0m s-color-border-warning-default -> use 's-color-border-warning'",
  "[93m[DEPRECATED][0m s-color-border-warning-subtle-default -> use 's-color-border-warning-subtle'",
  "[93m[DEPRECATED][0m s-color-border-info-default -> use 's-color-border-info'",
  "[93m[DEPRECATED][0m s-color-border-info-subtle-default -> use 's-color-border-info-subtle'",
  "[93m[DEPRECATED][0m s-color-icon-default -> use 's-color-icon'",
  "[93m[DEPRECATED][0m s-color-icon-subtle-default -> use 's-color-icon-subtle'",
  "[93m[DEPRECATED][0m s-color-text-default -> use 's-color-text'",
]
`;
